Library/Homebrew/extend/os/mac/formula.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class Formula
+  class << self
+    def uses_from_macos(dep, **args)
+      depends_on(dep) if add_mac_dependency?(args)
+    end
+
+    private
+
+    def add_mac_dependency?(args)
+      args.each { |key, version| args[key] = OS::Mac::Version.from_symbol(version) }
+
+      args.blank? ||
+        args[:before] && OS::Mac.version < args[:before] ||
+        args[:after] && OS::Mac.version >= args[:after]
```

Comment:
I suggest
```ruby
(!args[:after] || OS::Mac.version >= OS::Mac::Version.from_symbol(args[:after])) &&
(!args[:before] || OS::Mac.version < OS::Mac::Version.from_symbol(args[:before]))
```
or equivalently
```ruby
return false if args[:after] && OS::Mac.version < OS::Mac::Version.from_symbol(args[:after])
return false if args[:before] && OS::Mac.version >= OS::Mac::Version.from_symbol(args[:before])
return true
```
This allows specifying both `:before` and `:after`, which I know @MikeMcQuaid said that feature is not a priority. The current implementation however does weird stuff if both `:before` and `:after` are specified. If you keep the current code, please add a check that only one of the two is specified.

Reply:
I think we can mix everything and do:

```rb
def add_mac_dependency?(args)
  args.each { |key, version| args[key] = OS::Mac::Version.from_symbol(version) }

  return false if args[:after] && OS::Mac.version < args[:after]
  return false if args[:before] && OS::Mac.version >= args[:before]
  return true
end
```

to dry up the `OS::Mac::Version.from_symbol`